modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -297,14 +300,22 @@ private void declareMessageSource(HasSourceDeclarer declarer, Class<? extends So
                 .whichReturns(typeLoader.load(sourceGenerics.get(0)))
                 .withAttributesOfType(typeLoader.load(sourceGenerics.get(1)))
                 .withExceptionEnricherFactory(getExceptionEnricherFactory(sourceType))
-                .withModelProperty(new ImplementingTypeModelProperty(sourceType));
+                .withModelProperty(new ImplementingTypeModelProperty(sourceType))
+                .withMetadataResolverFactory(getMetadataResolverFactory(extensionType, sourceType));
+
+        declareSingleParameters(getParameterFields(sourceType), source, (ModelPropertyContributor) MuleExtensionAnnotationParser::parseMetadataAnnotations);
+        declareParameterGroups(sourceType, source);
 
-        declareAnnotatedParameters(sourceType, source);
     }
 
     private void declareAnnotatedParameters(Class<?> annotatedType, ParameterizedDeclarer parameterDeclarer)
     {
         declareSingleParameters(getParameterFields(annotatedType), parameterDeclarer);
+        declareParameterGroups(annotatedType, parameterDeclarer);
+    }
+
+    private void declareParameterGroups(Class<?> annotatedType, ParameterizedDeclarer parameterDeclarer)
```

Comment:
I group the logic to be more declarative and re use it in the Source parameters declaration